src/lepton/jpgcoder.cc

```diff
@@ -1648,6 +1648,31 @@ class CachingDecoderReader : public DecoderReader
                               size == 0 ? JpegError::errEOF() : JpegError::nil());
     }
 };
+
+
+// Implementation of Sirikata::DecoderReader API, using the provided std::vector
+// as input data stream
+class VectorDecoderReader : public DecoderReader {
+    const std::vector<uint8_t> &input_data;     // buffer holding entire stream contents
```

Comment:
Why not just `buffer` and `position`?

Reply:
Do you mean variable names and/or types (uint8_t* instead of vector) ?

Reply:
names